refactor: update `blas/ext/base/grev` to follow current project conventions

headlessNode opened the pull request:
## Description

> What is the purpose of this pull request?

This pull request:

-   update the package to follow current project conventions

## Related Issues

> Does this pull request have any related issues?

No. 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

headlessNode committed:
refactor: update to follow current project conventions

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: passed
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: passed
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: na
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

headlessNode committed:
bench: fix import path

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

Planeshifter commented on lib/node_modules/@stdlib/blas/ext/base/grev/benchmark/benchmark.js:
Can we please change this back to a generic array instead of `Float64Array`.

```suggestion
	var x = gfillBy( len, new Array( len ), 1, uniform( -100, 100 ) );
```

Planeshifter commented on lib/node_modules/@stdlib/blas/ext/base/grev/benchmark/benchmark.ndarray.js:
```suggestion
	var x = gfillBy( len, new Array( len ), 1, uniform( -100, 100 ) );
```

headlessNode committed:
refactor: apply code review suggestion

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

---
type: pre_push_report
description: Results of running various checks prior to pushing changes.
report:
  - task: run_javascript_examples
    status: na
  - task: run_c_examples
    status: na
  - task: run_cpp_examples
    status: na
  - task: run_javascript_readme_examples
    status: na
  - task: run_c_benchmarks
    status: na
  - task: run_cpp_benchmarks
    status: na
  - task: run_fortran_benchmarks
    status: na
  - task: run_javascript_benchmarks
    status: passed
  - task: run_julia_benchmarks
    status: na
  - task: run_python_benchmarks
    status: na
  - task: run_r_benchmarks
    status: na
  - task: run_javascript_tests
    status: na
---

headlessNode commented:
@Planeshifter Changes applied.

Planeshifter reviewed (approved):
Great, thanks!

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/grev/benchmark/benchmark.ndarray.js:
@Planeshifter This wasn't the correct change. If we wanted generic arrays, then we should have used `@stdlib/array/base/zeros` to ensure that the resultant array is not in dictionary mode.

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/grev/benchmark/benchmark.js:
Same comment.

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/grev/benchmark/benchmark.ndarray.js:
That's problematic, especially for large arrays, as it can tank performance, thus throwing off benchmark results.

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/grev/benchmark/benchmark.ndarray.js:
@headlessNode Would you mind submitting a follow-up PR addressing this?

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/grev/benchmark/benchmark.ndarray.js:
This would also be a good lint rule. Namely, we should disallow `new Array(...)`.

headlessNode commented on lib/node_modules/@stdlib/blas/ext/base/grev/benchmark/benchmark.ndarray.js:
@kgryte I actually did use the `@stdlib/array/base/zeros`, see next commit.

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/grev/benchmark/benchmark.ndarray.js:
@headlessNode One step ahead of me!

kgryte commented on lib/node_modules/@stdlib/blas/ext/base/grev/benchmark/benchmark.ndarray.js:
I should have looked at the final code, rather than just relying on the e-mail thread of discussions.
